src/test/kotlin/com/terraformation/backend/funder/db/FundingEntityStoreTest.kt

```diff
@@ -123,4 +128,64 @@ class FundingEntityStoreTest : DatabaseTest(), RunsAsUser {
         actualEntities[2].projects,
         "Third entity should have both projects")
   }
+
+  @Test
+  fun `fetchByProjectId requires user to be able to read funding entities`() {
+    every { user.canReadFundingEntities() } returns false
+
+    assertThrows<AccessDeniedException> { store.fetchByProjectId(ProjectId(1)) }
+  }
+
+  @Test
+  fun `fetchByProjectId requires user to be able to read project`() {
+    insertOrganization()
+    val projectId = insertProject()
+    every { user.canReadProject(projectId) } returns false
+
+    assertThrows<ProjectNotFoundException> { store.fetchByProjectId(projectId) }
+  }
+
+  @Test
+  fun `fetchByProjectId returns empty list when no funding entities are attached to project`() {
+    insertOrganization()
+    val fundingEntity1 = insertFundingEntity()
+    val projectId1 = insertProject()
+    val projectId2 = insertProject()
+    insertFundingEntityProject(fundingEntity1, projectId2)
+
+    assertTrue(
+        store.fetchByProjectId(projectId1).isEmpty(),
+        "Project should not be connected to any entities")
```

Comment:
For this kind of thing, I prefer `assertEquals(emptyList<Whatever>(), store.fetchWhatever(), "Message")` so the assertion failure message will show what was fetched if the list isn't empty.